src/RabbitMQ/AmqpSyncPublisher.php

```diff
@@ -0,0 +1,83 @@
+<?php
+namespace Burrow\RabbitMQ;
+
+use PhpAmqpLib\Channel\AMQPChannel;
+use PhpAmqpLib\Connection\AMQPConnection;
+use PhpAmqpLib\Message\AMQPMessage;
+use Burrow\QueuePublisher;
+
+class AmqpSyncPublisher extends AbstractAmqpPublisher implements QueuePublisher
+{
+    /**
+     * @var string
+     */
+    private $callbackQueue;
+
+    /**
+     * @var string
+     */
+    private $correlationId;
+
+    /**
+     * @var string
+     */
+    private $response;
+
+    /**
+     * Constructor
+     *
+     * @param string $host
+     * @param string $port
+     * @param string $user
+     * @param string $pass
+     * @param string $exchangeName
+     */
+    public function __construct($host, $port, $user, $pass, $exchangeName)
+    {
+        parent::__construct($host, $port, $user, $pass, $exchangeName);
+
+        $self = $this;
+        list($this->callbackQueue, ,) = $this->channel->queue_declare('', false, false, true, false);
+        $this->channel->basic_consume(
+            $this->callbackQueue, '', false, false, false, false, function (AMQPMessage $message) use ($self) {
+                if($message->get('correlation_id') == $self->correlationId) {
+                    $self->response = unserialize($message->body);
+                }
+            }
+        );
+    }
+
+    /**
+     * Publish a message on the queue
+     *
+     * @param string $data
+     * @param string $routingKey
+     *
+     * @return mixed|null|void
+     */
+    public function publish($data, $routingKey = '')
+    {
+        $this->response = null;
+        $this->correlationId = uniqid();
+
+        parent::publish($data, $routingKey);
+
+        while(!$this->response) {
+            $this->channel->wait();
```

Comment:
I'm not sure any of this is going to work for me as I have to wait to send something back at the end of the method, and that's the reason I have to wait.

To prevent an infinite loop I can add a timeout, but i really don't see how i can delegate it to a callback.

Or maybe I looked at it a little bit too quickly and didn't catch how to do it.